lib/BC/Util.cpp

```diff
@@ -208,38 +207,41 @@ llvm::CallInst *AddTerminatingTailCall(llvm::BasicBlock *source_block,
 
 // Find a local variable defined in the entry block of the function. We use
 // this to find register variables.
-llvm::Value *FindVarInFunction(llvm::BasicBlock *block, std::string_view name,
-                               bool allow_failure) {
+std::pair<llvm::Value *, llvm::Type *>
+FindVarInFunction(llvm::BasicBlock *block, std::string_view name,
+                  bool allow_failure) {
   return FindVarInFunction(block->getParent(), name, allow_failure);
 }
 
 // Find a local variable defined in the entry block of the function. We use
 // this to find register variables.
-llvm::Value *FindVarInFunction(llvm::Function *function, std::string_view name_,
-                               bool allow_failure) {
+std::pair<llvm::Value *, llvm::Type *>
+FindVarInFunction(llvm::Function *function, std::string_view name_,
+                  bool allow_failure) {
+  // TODO(alex): Figure out how to to get types for the two cases below.
```

Comment:
> However, we may also be dealing with "local var" regs, e.g. BRANCH_TAKEN. Those are allocas for which the value type can be deduced

@pgoodman This part wasn't clear to me. What did you mean by deducing the type from allocas?

Reply:
If you cast the thing to an `AllocaInst *`, and it works, then from the alloca inst you can get the `llvm::Type *` for the allocated value.

Reply:
What if it isn't an `AllocaInst` though? Just skip it? That's what I'm intending to do but it seems like it will find less variables this way.

Reply:
[669166d](https://github.com/lifting-bits/remill/pull/596/commits/669166d5725e51dc045efbe6ae05fd6d61262175)

From my test program, I saw that it can either be an alloca or a GEP so I handled those two cases. I also didn't see the code ever hit the function arg branch so I just removed it entirely.

Reply:
👍